oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -15,7 +15,7 @@
 /**
  * Base type for Credentials using OAuth2.
  */
-public abstract class OAuth2Credentials extends Credentials {
+public class OAuth2Credentials extends Credentials {
```

Comment:
Make the zero-arg constructor protected, since it is guaranteed to produce poor results?